test/rulesets/Base/indexing.jl

```diff
@@ -3,48 +3,48 @@
         x = [1.0 2.0 3.0; 10.0 20.0 30.0]
 
         @testset "single element" begin
-            test_rrule(getindex, x, 2 ⊢ nothing)
-            test_rrule(getindex, x, 2 ⊢ nothing, 1 ⊢ nothing)
-            test_rrule(getindex, x, 2 ⊢ nothing, 2 ⊢ nothing)
+            test_rrule(getindex, x, 2)
+            test_rrule(getindex, x, 2, 1)
+            test_rrule(getindex, x, 2, 2)
 
-            test_rrule(getindex, x, CartesianIndex(2, 3) ⊢ nothing)
+            test_rrule(getindex, x, CartesianIndex(2, 3) ⊢ DoesNotExist())
         end
 
         @testset "slice/index postions" begin
-            test_rrule(getindex, x, 2:3 ⊢ nothing)
-            test_rrule(getindex, x, 3:-1:2 ⊢ nothing)
-            test_rrule(getindex, x, [3,2] ⊢ nothing)
-            test_rrule(getindex, x, [2,3] ⊢ nothing)
+            test_rrule(getindex, x, 2:3 ⊢ DoesNotExist())
+            test_rrule(getindex, x, 3:-1:2 ⊢ DoesNotExist())
+            test_rrule(getindex, x, [3,2] ⊢ DoesNotExist())
+            test_rrule(getindex, x, [2,3] ⊢ DoesNotExist())
 
-            test_rrule(getindex, x, 1:2 ⊢ nothing, 2:3 ⊢ nothing)
-            test_rrule(getindex, x, (:) ⊢ nothing, 2:3 ⊢ nothing)
+            test_rrule(getindex, x, 1:2 ⊢ DoesNotExist(), 2:3 ⊢ DoesNotExist())
+            test_rrule(getindex, x, (:) ⊢ DoesNotExist(), 2:3 ⊢ DoesNotExist())
 
-            test_rrule(getindex, x, 1:2 ⊢ nothing, 1 ⊢ nothing)
-            test_rrule(getindex, x, 1 ⊢ nothing, 1:2 ⊢ nothing)
+            test_rrule(getindex, x, 1:2 ⊢ DoesNotExist(), 1)
```

Comment:
should we have 
```
function rand_tangent(x::AbstractArray)
    dx = map(rand_tangent, x)
    return all(dxi -> dxi isa DoesNotExist, dx) ? DoesNotExist() : dx
end
```
or something else like that to condense multiple representations of zero